apps/src/code-studio/components/libraries/LibraryPublisher.jsx

```diff
@@ -305,6 +341,38 @@ export default class LibraryPublisher extends React.Component {
     );
   };
 
+  allFunctionsSelected = () => {
+    const {sourceFunctionList} = this.props.libraryDetails;
+    const {selectedFunctions} = this.state;
+
+    let allSelected = true;
+    sourceFunctionList.forEach(sourceFunction => {
+      if (
+        !selectedFunctions[sourceFunction.functionName] &&
+        this.isFunctionValid(sourceFunction)
+      ) {
+        allSelected = false;
+      }
+    });
+
+    return allSelected;
+  };
+
+  toggleAllFunctionsSelected = () => {
+    if (this.allFunctionsSelected()) {
+      this.setState({selectedFunctions: {}});
+    } else {
+      const {sourceFunctionList} = this.props.libraryDetails;
+      let selectedFunctions = {};
+      sourceFunctionList.forEach(sourceFunction => {
+        if (this.isFunctionValid(sourceFunction)) {
+          selectedFunctions[sourceFunction.functionName] = true;
+        }
```

Comment:
Possible simplification of logic (up to you)
`selectedFunctions[sourceFunction.functionName] = this.isFunctionValid(sourceFunction)`